src/simplnx/Common/Ray.hpp

```diff
@@ -216,5 +240,26 @@ class Ray
   PointType m_Origin;
   ZXZEulerType m_Angle;
   LengthType m_Length;
+
+  // Optional caching for speed
+  PointType m_Endpoint = {};
```

Comment:
Right now all `Ray` will use space for the endpoint even if they aren't caching the value. You can do something similar to `Result` to have different members depending on the template arguments.